### DISCUSSION TITLE
Component proposal: Step indicator track-status variant

### DISCUSSION THREAD

**Discussion by jcandan** [uswds/uswds#5960] [Proposals]
### Component name

[Step indicator](https://designsystem.digital.gov/components/step-indicator)

### Component description

For publication or submission workflows, when a step has met some criteria, the step should indicate its status: incomplete or complete.

![image](https://github.com/uswds/uswds/assets/2332648/c93cd278-0776-45c6-9914-65733f3356e0)

### Why USWDS needs this component

Government websites often require a visitor to save their work to return later and complete the form. Indicating which steps are complete vs incomplete would be a positive user experience.

**Comment by jcandan** [comment]
It would be nice to see this work in conjunction with https://github.com/uswds/uswds/discussions/5961.

**Comment by jcandan** [comment]
I'd imagine we could simply use the Success and Warning colors to indicate a step's status as complete or incomplete, with darker or lighter variants for current steps.

Would a checkmark or other icon be prudent?

**Comment by jcandan** [comment]
Something like this

![image](https://github.com/uswds/uswds/assets/2332648/c93cd278-0776-45c6-9914-65733f3356e0)

**Comment by jcandan** [comment]
Here are some examples replacing the complete counters with check marks:

<img width="685" alt="image" src="https://github.com/user-attachments/assets/62b2a1ce-a56f-437d-af34-fb0aae30db20">

<img width="701" alt="image" src="https://github.com/user-attachments/assets/0981eddb-e3c7-4e49-87c0-483ceead5f15">

However, note that I am using the generally available CSS Entity rather than USWDS check icon.

```css
li.usa-step-indicator__segment.usa-step-indicator__segment--complete::before {
    content: "\2713";
    background-color: #5e9f69;
}
```

I was unable to do this via a URL function.

```
li.usa-step-indicator__segment.usa-step-indicator__segment--complete::before {
    content: url("https://designsystem.digital.gov/assets/img/sprite.svg#check");
}
```

I believe that USWDS Icons will need to provide its own set of CSS Entities for this to work.

**Reply by jcandan** [reply]
It seems Font Awesome uses `Object.defineProperty()` and Bootstrap uses `nfroidure/svgicons2svgfont` to create their custom CSS Entities.

References:
- https://github.com/search?q=org%3Atwbs+f38f&type=code
- https://github.com/search?q=repo%3Atwbs%2Ficons%20bootstrap-icons-map&type=code
- https://github.com/twbs/fantasticon/blob/master/src/generators/asset-types/svg.ts#L33
- https://github.com/search?q=repo%3AFortAwesome%2FFont-Awesome%20f000&type=code

This may warrant its own component proposal.

Does anyone have an alternative solution for using the USWDS icons in CSS?

**Comment by jcandan** [comment]
CC @mejiaj @mahoneycm

**Comment by jcklpe** [comment]
As per [this comment](https://github.com/uswds/uswds/discussions/5994#discussioncomment-10329773) by @brunerae  I believe requests to changes/extensions to existing components are supposed to be handled in an issue ticket here: https://github.com/uswds/uswds/issues/new/choose
